.github/workflows/scheduled.yml

```diff
@@ -117,10 +113,73 @@ jobs:
 
       - run: cargo test --release -p sncast
 
+  verify-version:
+    name: Get current foundry version
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.validVersion.outputs.version }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get version from Cargo.toml
+        id: lookupVersion
+        uses: mikefarah/yq@bbdd97482f2d439126582a59689eb1c855944955
+        with:
+          cmd: yq -oy '.workspace.package.version' 'Cargo.toml'
+          
+      - name: Compare versions
+        id: validVersion
+        run: |
+          COMMIT_VERSION=${{ steps.lookupVersion.outputs.result }}
+          echo "Project version from this commit = $COMMIT_VERSION"
+          echo "version=$COMMIT_VERSION" >> "$GITHUB_OUTPUT"
+
+  build-plugin-binaries:
+    name: Build plugin binaries
+    if: github.event_name == 'workflow_dispatch' || github.event_name == 'schedule'
+    needs: verify-version
+    uses: ./.github/workflows/_build-plugin-binaries.yml
+    with:
+      overridden_plugin_version: ${{ needs.verify-version.outputs.version }}-test.${{ github.sha }}${{ github.run_id }}
+          
+  publish-plugin:
```

Comment:
We might want to connect to scarb team and discuss how to tackle dev registry cleanup, may be periodically. With this, we'll very soon bloat the storage, which can be costly.